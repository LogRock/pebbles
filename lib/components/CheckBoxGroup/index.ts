export { default } from "./CheckBoxGroup";
export type { CheckBoxGroupProps } from "./CheckBoxGroup";
