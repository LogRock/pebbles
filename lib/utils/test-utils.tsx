/* eslint-disable import/export */
import React from "react";
import { cleanup, render } from "@testing-library/react";
import { afterEach } from "vitest";
import { ThemeProvider } from "styled-components";
import { light as theme } from "../theme";

afterEach(() => {
  cleanup();
});

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => (
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    ),
    ...options,
  });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
// override render export
export { customRender as render };
