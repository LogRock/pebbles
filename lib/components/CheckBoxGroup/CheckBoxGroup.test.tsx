import React from "react";

import { vi } from "vitest";

import "@testing-library/jest-dom";
// we should also import screen, userEvent, etc from this file.
import { render, screen, fireEvent } from "../../utils/test-utils";
// import { render } from "@testing-library/react"

import CheckBoxGroup from "./CheckBoxGroup";

describe("CheckBoxGroup Component", () => {
  it("should match snapshot", () => {
    const checkBoxGroup = render(
      <CheckBoxGroup possibleValues={["A", "B", "C"]} values={["A", "C"]} />
    );

    expect(checkBoxGroup).toMatchSnapshot();
  });
  it("should show a helper message snapshot", () => {
    render(
      <CheckBoxGroup
        description="test"
        helper="you should provide a value"
        possibleValues={["A", "B", "C"]}
        values={["A", "C"]}
      />
    );
    expect(screen.getByText(/you should provide a value/)).toBeInTheDocument();
  });
  it("should select options to trigger change event", () => {
    const onChange = () => null;
    const onChangeMock = vi.fn().mockImplementation(onChange);

    render(
      <CheckBoxGroup
        description="test"
        helper="you should provide a value"
        possibleValues={["A-test", "B-test", "C-test"]}
        values={["C-test"]}
        onChange={onChangeMock}
      />
    );
    fireEvent.click(screen.getByLabelText("A-test"));
    expect(onChangeMock).toHaveBeenCalledWith(["C-test", "A-test"]);
  });
});
