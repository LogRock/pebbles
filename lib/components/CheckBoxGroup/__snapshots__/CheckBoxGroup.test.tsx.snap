// Vitest Snapshot v1

exports[`CheckBoxGroup Component > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gFqAkR cbhzNS"
      >
        <label
          class="sc-cPiKLX gQejGr"
          color="neutral"
          for="pebbles__input__1"
          type="overlineXSmall"
        >
           
        </label>
        <div
          class="sc-hmdomO jqEVRc"
        >
          <div
            style="padding: 10px 12px;"
          >
            <label
              class="sc-eDPEul kSKwvk"
              for="A"
            >
              <div
                class="sc-eldPxv aBlwZ"
              >
                A
              </div>
              <input
                checked=""
                class="sc-fPXMVe zzeoB"
                id="A"
                type="checkbox"
              />
              <span
                class="sc-dLMFU gCsEgs"
              />
            </label>
            <label
              class="sc-eDPEul kSKwvk"
              for="B"
            >
              <div
                class="sc-eldPxv aBlwZ"
              >
                B
              </div>
              <input
                class="sc-fPXMVe zzeoB"
                id="B"
                type="checkbox"
              />
              <span
                class="sc-dLMFU gCsEgs"
              />
            </label>
            <label
              class="sc-eDPEul kSKwvk"
              for="C"
            >
              <div
                class="sc-eldPxv aBlwZ"
              >
                C
              </div>
              <input
                checked=""
                class="sc-fPXMVe zzeoB"
                id="C"
                type="checkbox"
              />
              <span
                class="sc-dLMFU gCsEgs"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gFqAkR cbhzNS"
    >
      <label
        class="sc-cPiKLX gQejGr"
        color="neutral"
        for="pebbles__input__1"
        type="overlineXSmall"
      >
         
      </label>
      <div
        class="sc-hmdomO jqEVRc"
      >
        <div
          style="padding: 10px 12px;"
        >
          <label
            class="sc-eDPEul kSKwvk"
            for="A"
          >
            <div
              class="sc-eldPxv aBlwZ"
            >
              A
            </div>
            <input
              checked=""
              class="sc-fPXMVe zzeoB"
              id="A"
              type="checkbox"
            />
            <span
              class="sc-dLMFU gCsEgs"
            />
          </label>
          <label
            class="sc-eDPEul kSKwvk"
            for="B"
          >
            <div
              class="sc-eldPxv aBlwZ"
            >
              B
            </div>
            <input
              class="sc-fPXMVe zzeoB"
              id="B"
              type="checkbox"
            />
            <span
              class="sc-dLMFU gCsEgs"
            />
          </label>
          <label
            class="sc-eDPEul kSKwvk"
            for="C"
          >
            <div
              class="sc-eldPxv aBlwZ"
            >
              C
            </div>
            <input
              checked=""
              class="sc-fPXMVe zzeoB"
              id="C"
              type="checkbox"
            />
            <span
              class="sc-dLMFU gCsEgs"
            />
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
